source/classification1.md

```diff
@@ -1159,68 +1100,74 @@ So far, we have built a preprocessor so that each of the predictors have a mean
 ```{index} scikit-learn; fit, scikit-learn; transform
 ```
 
-You can now see that the recipe includes a scaling and centering step for all predictor variables.
-Note that when you add a step to a `ColumnTransformer`, you must specify what columns to apply the step to.
-Here we specified that `StandardScaler` should be applied to 
-all predictor variables.
-
-```{index} see: fit, transform, fit_transform; scikit-learn
-```
-
-At this point, the data are not yet scaled and centered. To actually scale and center 
-the data, we need to call `fit` and `transform` on the unscaled data ( can be combined into `fit_transform`).
+You can see that the preprocessor includes a single standardization step
+that is applied to the `Area` and `Smoothness` columns. 
+Note that here we specified which columns to apply the preprocessing step to 
+by individual names; this approach can become quite difficult, e.g., when we have many
+predictor variables. Rather than writing out the column names individually,
+we can instead use the 
+[`make_column_selector`](https://scikit-learn.org/stable/modules/generated/sklearn.compose.make_column_selector.html#sklearn.compose.make_column_selector) function. For example, if we wanted to standardize all *numerical* predictors,
+we would use `make_column_selector` and specify the `dtype_include` argument to be `np.number`
+(from the `numpy` package). This creates a preprocessor equivalent to the one we
+created previously.
 
 ```{code-cell} ipython3
-:tags: [remove-cell]
+import numpy as np
+from sklearn.compose import make_column_selector
 
-# So far, there is not much in the recipe; just a statement about the number of targets
-# and predictors. Let's add 
-# scaling (`step_scale`) \index{recipe!step\_scale} and 
-# centering (`step_center`) \index{recipe!step\_center} steps for 
-# all of the predictors so that they each have a mean of 0 and standard deviation of 1.
-# Note that `tidyverse` actually provides `step_normalize`, which does both centering and scaling in
-# a single recipe step; in this book we will keep `step_scale` and `step_center` separate
-# to emphasize conceptually that there are two steps happening. 
-# The `prep` function finalizes the recipe by using the data (here, `unscaled_cancer`)  \index{tidymodels!prep}\index{prep|see{tidymodels}}
-# to compute anything necessary to run the recipe (in this case, the column means and standard
-# deviations):
+preprocessor = make_column_transformer(
+           (StandardScaler(), make_column_selector(dtype_include=np.number)),
+      )
```

Comment:
```suggestion
)
```

Reply:
In case this large indentation on open params happens because of the default vim settings, which was something that was driving me nuts for a long while, you can set `let g:pyindent_open_paren = 'shiftwidth()'` as per https://stackoverflow.com/questions/23016531/how-to-make-vim-only-insert-4-spaces-one-indent-level-for-the-next-line-after/74711118#74711118